### PR TITLE
cli: Support user project registration

### PR DESCRIPTION
**Pull request by NunoAlexandre**
Closes https://github.com/radicle-dev/radicle-registry/issues/491

### CODE REVIEW

File: cli/src/command/project.rs [hunk 1]
```diff
@@ -143,8 +156,56 @@ impl CommandT for Register {
...
             "✓ Project {}.{} registered in block {}",
-            self.project_name, self.org_id, project_registered.block,
+            self.project_name, project_domain, project_registered.block,
         );
         Ok(())
     }
 }
+
+#[derive(Clone, Eq, PartialEq, Debug, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "lowercase")]
```
Comment (geigerzaehler) [inline cli/src/command/project.rs:166]: Is there a reason why we implement `Deserialize` and `Serialize` for this type?

File: cli/src/command/project.rs [hunk 2]
```diff
@@ -143,8 +156,56 @@ impl CommandT for Register {
...
+}
+
+impl FromStr for DomainType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "org" => Ok(Self::Org),
+            "user" => Ok(Self::User),
+            _ => Err("Invalid DomainType".to_string()),
```
Comment (geigerzaehler) [inline cli/src/command/project.rs:179]: This is the message that is displayed to the user when they use a wrong domain. To make the message more helpful it should avoid code terminology (`DomainType`), repeat the input string, and list the possibilities.
```suggestion
            input => Err(format!("Invalid domain type \"{}\". Must be either org or user.", value)),
```

Comment (NunoAlexandre) [reply cli/src/command/project.rs:179]: It actually isn't given that we define the possible values in the struct opt construct. This case should therefore never happen and the user should never see that error message.

File: cli/src/command/project.rs (hunk 1 above)
Comment (NunoAlexandre) [reply cli/src/command/project.rs:166]: My bad, forgot to remove it. I had copy pasted it from another location.

File: cli/src/command/project.rs (hunk 2 above)
Comment (NunoAlexandre) [reply cli/src/command/project.rs:179]: To be more exact, the error users get is:

```
error: 'ORG' isn't a valid value for '<domain-type>'
        [possible values: org, user]
```

File: cli/src/command/project.rs [hunk 3]
```diff
@@ -143,8 +156,55 @@ impl CommandT for Register {
...
             "✓ Project {}.{} registered in block {}",
-            self.project_name, self.org_id, project_registered.block,
+            self.project_name, project_domain, project_registered.block,
         );
         Ok(())
     }
 }
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+enum DomainType {
```
Comment (CodeSandwich) [inline cli/src/command/project.rs:166]: Have you considered using https://docs.rs/clap/2.33.1/clap/macro.arg_enum.html?

Comment (CodeSandwich) [reply cli/src/command/project.rs]: Cool!

Comment (NunoAlexandre) [reply cli/src/command/project.rs]: TIL. Introduced it now. It implements `FromStr` and `Display` out of the box and dry's the `possible_values` property :+1:

**Pull request merged**